### PR TITLE
(PDB-1948) Use wire-v6 of reports in testutils

### PR DESCRIPTION
**Pull request by ajroetker**
This PR consolidates some of the reports munging we do in our testutils to a handful of functions and updates them all to use wire-v6 of reports.

### CODE REVIEW

**Comment by ajroetker** [comment]
Look like a transient ^ no change to the code since the last passing tests

**Comment by ajroetker** [comment]
@puppetlabs-jenkins test this please

File: test/puppetlabs/puppetdb/http/reports_test.clj [hunk 1]
```diff
@@ -421,25 +342,30 @@
...
 
-  (let [basic (:basic reports)
-        _ (store-example-report! basic (now))]
-
+  (let [basic (:basic reports)]
+    (store-example-report! basic (now))
     (testing "should return all reports for a certname"
-      (let [result (query-result method "/v4/environments/DEV/reports" ["=" "certname" (:certname basic)] {} munge-reports-for-comparison)]
+      (let [result (query-result method
+                                 "/v4/environments/DEV/reports" ["=" "certname" (:certname basic)] {}
```
Comment (wkalt) [inline test/puppetlabs/puppetdb/http/reports_test.clj]: might as well pop the query onto it's own line if we're just rearranging here

File: test/puppetlabs/puppetdb/testutils/events.clj [hunk 2]
```diff
@@ -11,24 +11,10 @@
...
-(defn environment [resource-event report version]
-  (if (= :v4 version)
-    (assoc-when resource-event :environment (:environment report))
-    (dissoc resource-event :environment)))
-
 (defn expected-resource-event
   "Given a resource event from the example data, plus a report hash, coerce the
   event into the format that we expect to be returned from a real query."
-  [version example-resource-event report]
+  [example-resource-event report]
```
Comment (wkalt) [inline test/puppetlabs/puppetdb/testutils/events.clj]: the version argument seems like it's bound to get reinserted everywhere down the road -- not advocating for or against it but that's the reason we've left it in in a lot of places.

Comment (wkalt) [inline test/puppetlabs/puppetdb/testutils/events.clj]: meh thinking about it more this probably won't happen for some time. removing it seems fine.

File: test/puppetlabs/puppetdb/testutils/reports.clj [hunk 3]
```diff
@@ -130,13 +101,19 @@
...
-                  munge-example-report-for-storage))))
+  [report]
+  (-> report
+      keywordize-keys
+      (update-in [:resource_events :data]
+                 (comp (partial sort-by
+                                #(mapv % [:timestamp :resource_type :resource_title :property]))
+                       (partial map
+                                (fn [resource-event]
+                                  (update resource-event :timestamp time-coerce/to-string)))))
```
Comment (wkalt) [inline test/puppetlabs/puppetdb/testutils/reports.clj]: the last few lines here seem a bit hairier than what we had. splitting out a couple more functions might make it clearer

**Comment by wkalt** [comment]
just a couple of minor comments -- this seems pretty straightforward and I'm generally +1

File: test/puppetlabs/puppetdb/testutils/reports.clj (hunk 3 above)
Comment (ajroetker) [inline test/puppetlabs/puppetdb/testutils/reports.clj]: Agreed, it was mostly trying to merge a bunch of different functionality together

**Pull request merged**